### PR TITLE
node: add missing ClientRequest events

### PR DESCRIPTION
**Pull request by clshortfuse**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/http.html#http_class_http_clientrequest
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

See also: https://github.com/nodejs/node/blob/master/lib/_http_client.js

### CODE REVIEW

**Comment by clshortfuse** [comment]
Just realized this isn't the right way to do this. I believe I have to copy/paste all the events within `Writable`. Seems like a bit redundant, but that's how `Http2Stream` appears to do it.

File: types/node/index.d.ts [hunk 1]
```diff
@@ -1283,6 +1283,137 @@ declare module "http" {
...
+
+        addListener(event: "abort", listener: () => void): this;
+        addListener(event: "connect", listener: (response: IncomingMessage, socket: net.Socket, head: Buffer) => void): this;
+        addListener(event: "continue", listener: () => void): this;
+        addListener(event: "information", listener: (information: { statusCode: number }) => void): this;
+        addListener(event: "response", listener: (response: IncomingMessage) => void): this;
+        addListener(event: "socket", listener: (socket: net.Socket) => void): this;
+        addListener(event: "timeout", listener: () => void): this;
+        addListener(event: "upgrade", listener: (response: IncomingMessage, socket: net.Socket, head: Buffer) => void): this;
+        addListener(event: "close", listener: () => void): this;
```
Comment (SimonSchick) [inline types/node/index.d.ts]: `close, drain, error, finish, unpipe` are already defined in the base class, remove these please.

Comment (clshortfuse) [reply types/node/index.d.ts]: I didn't have them originally, then realized that typescript won't read the base classes. 

The `addListener(event: string | symbol, listener: (...args: any[]) => void): this;` will override any base class. If I don't include that line, then there's a property/signature mismatch.

The same is done with `Http2Stream`.

**Comment by clshortfuse** [comment]
I guess. I'll open up another PR...

**Pull request closed without merging**